src/Processors/Executors/ExecutionThreadContext.cpp

```diff
@@ -67,7 +68,10 @@ static void executeJob(ExecutingGraph::Node * node, ReadProgressCallback * read_
     catch (Exception & exception)
     {
         if (checkCanAddAdditionalInfoToException(exception))
+        {
```

Comment:
Could you please explain what's going on?

Are multiple different threads throwing the same exception? How / why?

Do we need locking or do could we just copy the exception and change it, that is, change it to `catch (Exception exception)` instead of using a reference

Reply:
Uhmm, Looking at `rethrowExceptionIfHas` it seems that we should fix it there instead by doing a proper handling of the atomic variable (`has_exception`) with `compare_exchange_weak` or something similar. With the current code there is a race condition. WDYT?

Reply:
Hm, I meant `ExecutionThreadContext::rethrowExceptionIfHas()` (I don't see a race condition there), not `PushingAsyncPipelineExecutor::Data::rethrowExceptionIfHas()` (this looks suspicious). But I'm not familiar with this code and not sure which one actually throws